src/kernel/binary_reduce_common.h

```diff
@@ -0,0 +1,450 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file kernel/binary_reduce_common.h
+ * \brief Common utilities for binary reduce operation.
+ */
+#ifndef DGL_KERNEL_BINARY_REDUCE_COMMON_H_
+#define DGL_KERNEL_BINARY_REDUCE_COMMON_H_
+
+#include <dgl/runtime/ndarray.h>
+
+#include <limits>
+#include <string>
+
+#include "./common.h"
+
+namespace dgl {
+namespace kernel {
+namespace binary_op {
+
+static const char kReduceSum[] = "sum";
+static const char kReduceMax[] = "max";
+static const char kReduceMin[] = "min";
+static const char kReduceMean[] = "mean";
+static const char kReduceProd[] = "prod";
+static const char kReduceNone[] = "none";
+
+static const char kAdd[] = "add";
+static const char kSub[] = "sub";
+static const char kMul[] = "mul";
+static const char kDiv[] = "div";
+static const char kUseLhs[] = "use_lhs";
+
+enum Target {
+  kSrc = 0,
+  kDst,
+  kEdge,
+  kNone,
+};
```

Comment:
what does this mean? please write some comments.

Reply:
More comments. Pls checkout.